### PR TITLE
[feature] Complete config across imports | #BAZEL-1186

### PR DESCRIPTION
**Pull request by mtoader**
Also fix import resolution of sibling file (e.g. import `.xxx.bazelrc`)

### CODE REVIEW

**Comment by mtoader** [comment]
@mpiekutowski can i have some eyes on this too when possible ?

**Comment by mtoader** [comment]
@tpasternak can you take a look ?

**Comment by tpasternak** [comment]
Sorry for that, but I was OOO for two weeks and now I'm flooded with housekeeping tasks in he native plugin

File: plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  override fun process(currentElement: BazelrcElement): Boolean {
+    if (currentElement == inputElement) return true
+    processElement(currentElement)
+    return true
+  }
+
+  fun processFile(file: BazelrcFile?) {
+    file?.virtualFile?.apply {
+      seenFiles.add(this).ifTrue {
```
Comment (agluszak) [inline plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt:22]: `ifTrue` seems to come from some weird package, could you remove it please?

File: plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        file.imports.map(::processElement)
+        file.lines.map(::processElement)
+      }
+    }
+  }
+
+  fun processElement(current: BazelrcElement) =
+    when (current) {
+      is BazelrcLine -> processLine(current)
+      is org.jetbrains.bazel.languages.bazelrc.psi.BazelrcImport -> processImport(current)
```
Comment (agluszak) [inline plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt:32]:
```suggestion
      is BazelrcImport -> processImport(current)
```

File: plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      else -> {}
+    }
+
+  private fun processLine(line: BazelrcLine) {
+    (line.containingFile as? BazelrcFile)?.apply {
+      line.configName()?.let { results[it] = this }
+    }
+  }
+
+  private fun processImport(element: org.jetbrains.bazel.languages.bazelrc.psi.BazelrcImport) {
```
Comment (agluszak) [inline plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt:42]:
```suggestion
  private fun processImport(element: BazelrcImport) {
```

**Review by agluszak — CHANGES_REQUESTED** [review]

**Comment by agluszak** [comment]
Otherwise LGTM!

File: plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt (hunk 2 above)
Comment (mtoader) [reply plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt]: No idea how this remained. I saw idea doing some half work when moving code around ( non shortinging imports correctly) but this is just one branch of the when .. doesn't really make sense :)

Thanks for catching

File: plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt (hunk 1 above)
Comment (mtoader) [reply plugin-bazel/src/main/kotlin/org/jetbrains/bazel/languages/bazelrc/completion/BazelrcConfigDeclarationsProcessor.kt]: The package started with org.jetbrains :) that automatically doesn't make it weird :) Fixed.

**Review by agluszak — APPROVED** [review]

**Pull request merged**